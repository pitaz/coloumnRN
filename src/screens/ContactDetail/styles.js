import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  homeWrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 40,
    paddingBottom: 40,
  },
  pageWrapper: {
    flex: 1,
  },
  contentWrapper: {
    flex: 1,
    justifyContent: 'center',
    padding: 40,
  },
  avatar: {
    height: 100,
    width: 100,
    borderRadius: 75,
    backgroundColor: '#242038',
    marginRight: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  avatarText: {
    color: '#fff',
  },
  name: {
    padding: 20,
    fontSize: 20,
  },
  backText: {
    padding: 20,
  },
});
