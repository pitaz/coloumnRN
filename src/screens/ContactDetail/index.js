import React from 'react';
import {View, Text, Image, SafeAreaView, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';
import Card from '../../components/Card';
import {nameInitials} from '../../helpers/nameInitials';
import {styles} from './styles';

/**
 *
 * Contact details
 *
 * @param {object} props
 *
 * @returns {void}
 */
const ContactDetail = (props) => {
  const {navigation, route} = props;
  const contact = route.params;
  const renderView = () => (
    <Card>
      <View style={styles.homeWrapper}>
        {contact?.hasThumbnail ? (
          <Image source={{uri: contact?.thumbnailPath}} style={styles.avatar} />
        ) : (
          <View style={styles.avatar}>
            <Text style={styles.avatarText}>
              {nameInitials(contact.givenName, contact.familyName)}
            </Text>
          </View>
        )}
        <Text style={styles.name}>
          {`${contact.givenName} ${contact.familyName}`}
        </Text>
        <Text>{`${contact?.mobileType}: ${contact?.number}`}</Text>
      </View>
    </Card>
  );
  return (
    <SafeAreaView style={styles.pageWrapper}>
      <TouchableOpacity onPress={() => navigation.goBack()}>
        <Text style={styles.backText}>Back</Text>
      </TouchableOpacity>
      <View style={styles.contentWrapper}>{renderView()}</View>
    </SafeAreaView>
  );
};

ContactDetail.propTypes = {
  props: PropTypes.object,
};

export default ContactDetail;
