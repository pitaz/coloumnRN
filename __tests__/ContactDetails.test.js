import React from 'react';
import {render, cleanup} from '@testing-library/react-native';
import ContactDetail from '../src/screens/ContactDetail';

afterEach(cleanup);

describe('Home', () => {
  let rendered;
  const navigation = {goBack: jest.fn()};
  const mockData = {
    givenName: 'Test name',
    familyName: 'Test Family name',
    thumbnailPath: '',
    mobileType: 'home',
    number: '0998777776666',
    hasThumbnail: false,
  };
  const route = {params: {}};
  beforeEach(() => {
    rendered = render(<ContactDetail navigation={navigation} route={route} />);
  });
  it('should match snapshot', () => {
    rendered.toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders back button', () => {
    const route2 = {params: {...mockData}};
    rendered = render(<ContactDetail navigation={navigation} route={route2} />);
    expect(rendered.queryByText('Back')).not.toBeNull();
  });
});
