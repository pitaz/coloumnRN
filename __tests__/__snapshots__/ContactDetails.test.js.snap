// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "padding": 20,
        }
      }
    >
      Back
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "padding": 40,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 14,
            "elevation": 1,
            "marginBottom": 37.5,
            "marginLeft": 37.5,
            "marginRight": 37.5,
            "marginTop": 37.5,
            "paddingBottom": 37.5,
            "paddingLeft": 15,
            "paddingRight": 52.5,
            "paddingTop": 37.5,
            "shadowColor": "rgba(0, 0, 0, 0.15)",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 15,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingBottom": 40,
            "paddingTop": 40,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#242038",
              "borderRadius": 75,
              "height": 100,
              "justifyContent": "center",
              "marginRight": 10,
              "width": 100,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            undefinedundefined
          </Text>
        </View>
        <Text
          style={
            Object {
              "fontSize": 20,
              "padding": 20,
            }
          }
        >
          undefined undefined
        </Text>
        <Text>
          undefined: undefined
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
